File: src/main/java/nuclearcoder/discordbot/command/manage/CmdManageBot.java
package nuclearcoder.discordbot.command.manage;

import nuclearcoder.discordbot.BotUtil;
import nuclearcoder.discordbot.NuclearBot;
import nuclearcoder.discordbot.command.Command;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import sx.blah.discord.handle.obj.IMessage;
import sx.blah.discord.handle.obj.IPrivateChannel;
import sx.blah.discord.handle.obj.IUser;
import sx.blah.discord.util.DiscordException;
import sx.blah.discord.util.RequestBuffer;

public class CmdManageBot implements Command {

    private static final Logger LOGGER = LoggerFactory.getLogger(CmdManageBot.class);

    @Override public void execute(final NuclearBot bot, final IMessage message,
            final String command, final String[] args)
    {
        final IUser user = message.getAuthor();

        if (BotUtil.isOperator(user.getID()))
        {
            RequestBuffer.request(() ->
            {
                try
                {
                    IPrivateChannel pmChannel = user.getOrCreatePMChannel();

                    if (command.equalsIgnoreCase("stop"))
                    {
                        bot.terminate(false);
                    }
                    else if (command.equalsIgnoreCase("restart"))
                    {
                        bot.terminate(true);
                    }
                    else if (command.equalsIgnoreCase("set_op"))
                    {
                        if (args.length > 2)
                        {
                            boolean setOp = Boolean.parseBoolean(args[1]);
                            String notification = "You are " + (setOp ? "now" : "no longer")
                                    + " operator for NuclearBot. :warning:";

                            StringBuilder sb = new StringBuilder();
                            sb.append(setOp ? "Set" : "Un-set");
                            sb.append(" to operator:");

                            for (IUser mentioned : message.getMentions())
                            {
                                // set op
                                BotUtil.setOperator(mentioned.getID(), setOp);
                                // notify new op
                                BotUtil.sendMessage(mentioned.getOrCreatePMChannel(), notification);
                                sb.append("\n- ");
                                sb.append(mentioned.mention());
                            }

                            BotUtil.sendMessage(pmChannel, sb.toString());
                            // notify executor
                        }
                    }
                }
                catch (DiscordException e)
                {
                    LOGGER.error("Error in bot management command:", e);
                }
            });
        }
    }

}

File: src/main/java/nukebot/command/manage/CmdManageBot.kt
package nukebot.command.manage

import nukebot.LOGGER
import nukebot.NuclearBot
import nukebot.command.Command
import nukebot.reloadModules
import nukebot.util.isOperator
import nukebot.util.rSendMessage
import nukebot.util.setOperator
import sx.blah.discord.handle.obj.IMessage
import sx.blah.discord.util.DiscordException
import sx.blah.discord.util.RequestBuffer

class CmdManageBot : Command {

    override fun execute(bot: NuclearBot, message: IMessage,
                         command: String, args: Array<String>) {
        val user = message.author

        if (isOperator(user.id)) {
            RequestBuffer.request {
                try {
                    val pmChannel = user.orCreatePMChannel

                    if (command.equals("stop", ignoreCase = true)) {
                        bot.terminate(false)
                    } else if (command.equals("restart", ignoreCase = true)) {
                        bot.terminate(true)
                    } else if (command.equals("setop", ignoreCase = true)) {
                        if (args.size > 2) {
                            val setOp = args[1].toBoolean()
                            val notification = "You are " + (if (setOp) "now" else "no longer") + " operator for NuclearBot. :warning: "

                            val sb = StringBuilder()
                            sb.append(if (setOp) "Set" else "Un-set")
                            sb.append(" to operator:")

                            for (mentioned in message.mentions) {
                                // set op
                                setOperator(mentioned.id, setOp)
                                // notify new op
                                mentioned.orCreatePMChannel.rSendMessage(notification)
                                sb.append("\n**")
                                sb.append(mentioned.name)
                                sb.append("**")
                            }

                            pmChannel.rSendMessage(sb.toString())
                            // notify executor
                        }
                    } else if (command.equals("reload", ignoreCase = true)) {
                        reloadModules(bot.moduleLoader)
                    }
                } catch (e: DiscordException) {
                    LOGGER.error("Error in bot management command:", e)
                }
            }
        }
    }

}
